packages/api/src/auth/auth.service.ts

```diff
@@ -0,0 +1,32 @@
+import { ConflictException, Injectable } from '@nestjs/common';
+import { EventBus } from '@nestjs/cqrs';
+import { Prisma } from '@prisma/client';
+
+import { registerError } from '@coderscamp/shared/models/auth/register';
+import { pick } from '@coderscamp/shared/utils/object';
+
+import { AuthRepository } from './auth.repository';
+import { UserRegistrationCompletedEvent } from './events/user-registration-completed.event';
+import { hashPassword } from './local/local.utils';
+
+@Injectable()
+export class AuthService {
+  constructor(private readonly eventBus: EventBus, private readonly authRepository: AuthRepository) {}
+
+  async register(data: Prisma.UserRegistrationCreateInput & Prisma.AuthUserCreateInput) {
+    const current = await this.authRepository.findUserRegistration({ where: { email: data.email } });
+
+    if (current) {
```

Comment:
It may be true/false in line 19, but in 26, while we're creating user registration the state may be different. 
You can address it with database transaction isolation levels, or just do not check it in the code, but handle exception from database if email unique constraint may be violated.